bin/cron/create_rollup_tables

```diff
@@ -0,0 +1,155 @@
+
+# This script populates (after truncating) the contained_levels,
+# contained_level_answers, and the level_sources_multi_type tables.
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/only_one'
+
+TIME_NOW = DateTime.now.strftime('%F %T').freeze
+
+# @param unsanitized [String] the unsanitized string
+# @returns [String] the sanitized version of the string, with newlines stripped
+#   and double quotations escaped
+def sanitize_string_for_db(unsanitized)
+  unsanitized.gsub(/[\r\n]+/, '').gsub(/"/, '\"')
+end
+
+def main
+  # Truncate the existing tables.
+  %w(
+    contained_levels
+    contained_level_answers
+    level_sources_multi_types
+  ).each do |table_name|
+    ActiveRecord::Base.connection.execute(
+      "TRUNCATE #{table_name}"
+    )
+  end
+
+  # Build the data for the tables.
+  contained_levels = []
+  contained_level_answers = []
+  level_sources_multi_types = []
+
+  Level.where(type: 'LevelGroup').find_each do |level|
+    level.properties['pages'].each_with_index do |page, page_index|
+      page['levels'].each_with_index do |contained_level_name, position_index|
+        contained_level = Level.find_by_name(contained_level_name)
+        next unless contained_level.type == 'Multi'
+        contained_question = contained_level.properties['markdown']
+        unless contained_question
+          contained_question = contained_level.
+            properties['questions'].
+            first['text']
+        end
+        contained_levels << {
+          level_id: level.id,
+          contained_level_id: contained_level.id,
+          contained_level_type: contained_level.type,
+          contained_level_page: page_index,
+          contained_level_position: position_index,
+          contained_level_text: sanitize_string_for_db(contained_question)
+        }
+
+        answers = contained_level.properties['answers']
+        answers.each_with_index do |answer, answer_number|
+          contained_level_answers << {
+            contained_level_id: contained_level.id,
+            answer_number: answer_number,
+            answer_text: sanitize_string_for_db(answer['text']),
+            correct: answer['correct']
+          }
+        end
+      end
+    end
+  end
+
+  LevelSource.where(level_id: contained_levels.map{|row| row[1]}).
+    find_each do |level_source|
+    level_sources_multi_types << {
+      level_source_id: level_source.id,
+      level_id: level_source.level_id,
+      data: level_source.data,
+      md5: level_source.md5,
+      hidden: level_source.hidden
+    }
+  end
+
+  # Verify that the data size is sensible.
+  raise 'contained_levels too big' if contained_levels.size > 1_000
+  raise 'contained_level_answers too big' if contained_level_answers.size > 10_000
+  raise 'level_sources too big' if level_sources_multi_types.size > 10_000
+
+  # Write the data to the DB tables.
+  contained_levels.each do |contained_level_row|
+    ActiveRecord::Base.connection.execute(
```

Comment:
Given that this is (will be) happening weekly on production, seems prudent. Expect that change in a forthcoming commit.